autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/ui/credential/management/CredentialListSorter.kt

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2022 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.autofill.ui.credential.management
+
+import com.duckduckgo.autofill.domain.app.LoginCredentials
+import com.duckduckgo.di.scopes.FragmentScope
+import com.squareup.anvil.annotations.ContributesBinding
+import javax.inject.Inject
+
+interface CredentialListSorter {
+    fun sort(credentials: List<LoginCredentials>): List<LoginCredentials>
+}
+
+@ContributesBinding(FragmentScope::class)
+class CredentialListSorterByTitleAndDomain @Inject constructor(
```

Comment:
I'm not 100% happy with the logic in here. It feels more complicated than it should be, and I can't help but think having multiple sorting criteria being applied is the best way to do it (how I had it first) instead of manually combining title and domain together in a single sort criteria like I'm doing now.

However, the unit tests will be a good guide to testing improvements here so we can come back to improving this and use the unit tests as the spec.